File: Modulo3-ObjectiveC-Lorena/Exercicio12.m
//
//  main.m
//  Exercicio12-C
//
//  Created by Lorena Rodrigues Bruno on 06/09/2018.
//  Copyright © 2018 Lorena Rodrigues Bruno. All rights reserved.
//

#import <Foundation/Foundation.h>

void transporMatriz (int matriz[4][4], int matrizTransposta[4][4]) {
    int linha, coluna;
    for(linha=0; linha < 4; linha++) {
        for(coluna=0; coluna < 4; coluna++) {
            if (linha == coluna)
                matrizTransposta[linha][coluna] = matriz[linha][coluna];
            else
                matrizTransposta[linha][coluna] = matriz[coluna][linha];
        }
    }
}

void imprimeMatriz (int matriz[4][4]) {
    int linha, coluna;
    
    for(linha=0; linha < 4; linha++){
        for(coluna=0; coluna < 4; coluna++) {
            printf("%i ", matriz[linha][coluna]);
        }
        printf("\n");
    }
}

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        int matriz[4][4], matrizTransposta[4][4], linha, coluna, entrada;
        
        printf("Digite 16 valores: ");
        for(linha=0; linha<4; linha++) {
            for(coluna=0; coluna<4; coluna++)
            {
                scanf("%i", &entrada);
                matriz[linha][coluna] = entrada;
            }
        }
        
        transporMatriz(matriz, matrizTransposta);
        printf("Matriz: \n");
        imprimeMatriz(matriz);
        
        printf("Matriz transposta: \n");
        imprimeMatriz(matrizTransposta);
    }
    return 0;
}

File: Modulo3-Lorena/Exercicio12.swift
//
//  main.swift
//  Exercicio12
//
//  Created by Lorena Rodrigues Bruno on 27/08/2018.
//  Copyright © 2018 Lorena Rodrigues Bruno. All rights reserved.
//

import Foundation

var entrada : String? = String()
var matrizEntrada = [[Int]](repeating: [Int](repeating: 0, count: 4), count: 4)
var matrizTransposta = [[Int]](repeating: [Int](repeating: 0, count: 4), count: 4)
var tamanhoMatriz = matrizEntrada.count

func transporMatriz (matriz:[[Int]]) {
    for row in 0..<tamanhoMatriz {
        for column in 0..<tamanhoMatriz{
            if row == column {
                matrizTransposta[row][column] = matrizEntrada[row][column]
            } else {
                matrizTransposta[row][column] = matrizEntrada[column][row]
            }
        }
    }
    
    print("Matriz transposta: \n\(matrizTransposta)")
}


print("Digite 16 valores: ")


for row in 0..<tamanhoMatriz {
    for column in 0..<tamanhoMatriz{
        let aux : Int? = Int(readLine()!)
        matrizEntrada[row][column] = aux!
    }
}

print("\n\(matrizEntrada)\n")

transporMatriz(matriz: matrizEntrada)







